packages/interface/src/lib.ts

```diff
@@ -95,30 +81,29 @@ export interface UCANOptions {
 }
 
 export interface DelegationOptions<
-  C extends [Capability, ...Capability[]],
+  C extends Capabilities,
   A extends number = number
 > extends UCANOptions {
-  issuer: SigningAuthority<A>
-  audience: Identity
+  issuer: SigningPrincipal<A>
+  audience: Principal
   capabilities: C
+  proofs?: Proof[]
 }
 
-export interface Delegation<
-  C extends [Capability, ...Capability[]] = [Capability, ...Capability[]]
-> {
-  readonly root: Transport.Block<C>
-  readonly blocks: Map<string, Transport.Block>
+export interface Delegation<C extends Capabilities = Capabilities> {
+  readonly root: UCANBlock<C>
+  readonly blocks: Map<string, Block>
```

Comment:
```suggestion
  /**
    * Map of all the IPLD blocks that where included with this delegation DAG.
    * Usually this would be blocks corresponding to proofs, however it may
    * also contain other blocks e.g. things that `capabilities` or `facts` may
    * link.
    * It is not guaranteed to include all the blocks of this DAG, as it represents
    * a partial DAG of the delegation desired for transporting.
    *
    * Also note that map may contain blocks that are not part of this
    * delegation DAG. That is because `Delegation` is usually constructed as
    * view / selection over the CAR which may contain bunch of other blocks.
    */
  readonly blocks: Map<string, Block>
```

Reply:
@hugomrdias does this suggested comment helps ?